src/app/application.cpp

```diff
@@ -257,11 +257,9 @@ Application::Application(int &argc, char **argv)
     Logger::initInstance();
 
     const auto portableProfilePath = Path(QCoreApplication::applicationDirPath()) / DEFAULT_PORTABLE_MODE_PROFILE_DIR;
-    const bool portableModeEnabled = m_commandLineArgs.profileDir.isEmpty() && portableProfilePath.exists();
-
-    const Path profileDir = portableModeEnabled
-        ? portableProfilePath
-        : m_commandLineArgs.profileDir;
+    const bool portableModeEnabled = m_commandLineArgs.profileDir.isEmpty() && Utils::Fs::isDir(portableProfilePath)
```

Comment:
Shouldn't we at least inform the user about a suspicious situation if one is detected (for example, the "profile" is not a folder or does not have the appropriate permissions)?

Reply:
I don't feel strongly about it. I'll prefer leave this job for others.

Reply:
Wouldn't it just change one problem to another? Let's think about it:
If it encounters the fact that there is non-folder "profile", then this clearly (most likely) does not look like the case when the user intends to use it in portable mode, so we can safely ignore it and run it normally. But what if we have exactly the "profile" folder, but it has invalid permissions? It may already look like we have a portable configuration, but it is damaged for some reason, and the user hardly wants to get qBittorrent running normally without any warning and the ability to cancel it. It would be better if it just couldn't get started. Don't you think so?

Reply:
>But what if we have exactly the "profile" folder, but it has invalid permissions? It may already look like we have a portable configuration, but it is damaged for some reason, and the user hardly wants to get qBittorrent running normally without any warning and the ability to cancel it. It would be better if it just couldn't get started. Don't you think so?

From my viewpoint, this `profile` folder is an auto-detection. If it is usable then qbt use it. If not then too bad, qbt just ignores it. It doesn't warrant too many error message/reporting for it.
IIRC there is a commandline option for the user if he really wants to specify a specific profile folder.

Reply:
I'm not sure you understand what I'm worried about.
Just imagine, you insert a USB flash drive with a portable qBittorrent configuration and run it, but for some reason the "profile" folder does not have write permission on the current system (I don't know why, maybe because of security settings or the like). Would you expect qBittorrent to silently launch with all the undesirable consequences that follow, for example, creating files/folders where you would not want them?

Reply:
>Would you expect qBittorrent to silently launch with all the undesirable consequences that follow, for example, creating files/folders where you would not want them?

I would expect the user to use the profile commandline option to run qbt. Probably with a batch script which will be handy.
As said before, 'profile' folder is an auto-detection, at least this is how it was designed from the start IIRC (not my idea though).

Reply:
>So can we just leave isDir() check?

Don't understand, please rephrase.

Reply:
> > So can we just leave isDir() check?
> 
> Don't understand, please rephrase.

Don't check permissions. Only whether "profile" is directory.

Reply:
>Don't check permissions. Only whether "profile" is directory.

OK, done.

Reply:
> I would expect the user to use the profile commandline option to run qbt.

We are talking about users who want to do nothing more than create a "profile" folder next to the qBittorrent executable.

I'm just trying to reflect on real-world scenarios and come up with a compromise solution for different types of users.

Reply:
It seems that one of the factors influencing this problem is the use of a commonly used name for the portable profile folder. Maybe we should change it to something more personal like `qBittorrent_profile`.

Reply:
>It seems that one of the factors influencing this problem is the use of a commonly used name for the portable profile folder. Maybe we should change it to something more personal like qBittorrent_profile.

I wouldn't be surprised that linux users would have name clash with `profile`. Linux puts all executable under `/usr/bin` and it is easy to have another file named `profile` (as in the case of #20513). However for Windows it is very unlikely to be a problem as the executable usually live in its own dedicated folder and name clash is unlikely to happen.